src/OpenLoco/src/Windows/MapWindow.cpp

```diff
@@ -218,12 +223,668 @@ namespace OpenLoco::Ui::Windows::MapWindow
+                    case ElementType::industry:
+                        // Vanilla omits the ghost check, but I think it should be here anyway.
+                        if (!el.isGhost())
+                        {
+                            colour0 = colourFlash0 = PaletteIndex::index_0B;
+                            colourFlash1 = colourFlash0;
+                            colour1 = colour0;
+                        }
+                        break;
+
+                    case ElementType::track:
+                    case ElementType::station:
+                    case ElementType::road:
+                    {
+                        if (el.isGhost() || el.isAiAllocated())
+                            continue;
+
+                        auto owner = CompanyId::null;
+
+                        if (auto* stationEl = el.as<StationElement>())
+                        {
+                            auto station = StationManager::get(stationEl->stationId());
+                            owner = station->owner;
+                        }
+                        else if (auto* trackEl = el.as<TrackElement>())
+                        {
+                            owner = trackEl->owner();
+                        }
+                        else if (auto* roadEl = el.as<RoadElement>())
+                        {
+                            owner = roadEl->owner();
+                        }
+
+                        auto companyColour = CompanyManager::getCompanyColour(owner);
+                        colourFlash1 = colourFlash0 = colour1 = colour0 = Colours::getShade(companyColour, 5);
+                        if (_flashingItems & (1 << enumValue(owner)))
+                        {
+                            colourFlash1 = colourFlash0 = _flashColours[colour0];
+                        }
+                        break;
+                    }
+
+                    default:
+                        break;
+                };
+            }
+
+            mapPtr[0] = colour0;
+            mapPtr[1] = colour1;
+            mapAltPtr[0] = colourFlash0;
+            mapAltPtr[1] = colourFlash1;
+
+            pos += delta;
+            mapPtr += kRenderedMapWidth + 1;
+            mapAltPtr += kRenderedMapWidth + 1;
+        }
+
+        _drawMapRowIndex++;
+        if (_drawMapRowIndex > kMapColumns)
+            _drawMapRowIndex = 0;
+    }
+
     // 0x0046C544
-    static void sub_46C544(Window* self)
+    static void setMapPixels(const Window& self)
     {
-        registers regs;
-        regs.esi = X86Pointer(self);
-        call(0x0046C544, regs);
+        _flashingItems = self.var_854;
+        auto offset = _drawMapRowIndex * (kRenderedMapWidth - 1) + (kMapRows - 1);
+        auto* mapPtr = &_mapPixels[offset];
+        auto* mapAltPtr = &_mapAltPixels[offset];
+
+        Pos2 pos{};
+        Pos2 delta{};
+        switch (WindowManager::getCurrentRotation())
+        {
+            case 0:
```

Comment:
I'm surprised we don't have an enum for the rotation

Reply:
Yess, that would be a nice refactor.